jekyll/_cci2/config-intro.adoc

```diff
@@ -0,0 +1,332 @@
++ 
+. CircleCI uses your `config.yml` file to run the pipeline. You can see the output of the job in the CircleCI dashboard. A green tick denotes a successful pipeline; a red exclamation mark alerts you to a failure. Click on the job for more details.
++
+image::config-intro-part1-job.png[Successful build job]
++
+You should see your step - The First Step - with the the output of the commands:
++
+image:config-first-step.png[Successful step within job]
+
+NOTE: Although the `config.yml` syntax itself is straightforward, the indentation is more complicated. Incorrect indentation is the most common error. If you are experiencing problems with this example, check your indentation carefully, or copy and paste the sample code.
+
+== Part 2: Using code from your repo
+
+CircleCI provides several commands to simplify complex actions. In this example, you will use the `checkout` command. This command fetches the code from your git repo. Once you have retrieved that code, you can work with it in subsequent steps.
+
+If you have not done so already, create a `.circleci` directory in your project, add a `config.yml` file, and type or paste the code from the example below.
+
+If you already have a `config.yml` file from the example above, add another run step (lines 13-17) at the end. Make sure you observe the indentation.
+
+[source,yaml]
+----
+version: 2.1
+jobs:
+  build:
+    docker:
+      - image: cimg/base:2021.04
+    steps:
+      - checkout
+      - run:
+          name: The First Step
+          command: |
+            echo 'Hello World!'
+            echo 'This is the delivery pipeline'
+      - run:
+          name: The Second Step
+          command: |
+            ls -al
+            echo '^^^The files in your repo^^^'
+----
+
+These two small changes have significantly increased the functionality of your config file:
+
+* *Line 5:* This line now specifies a Docker image that supports git. `cimg/base:2021.04` is a small Ubuntu-based image for running basic jobs.
+* *Line 7:* The `checkout` command fetches the code from your git repo.
+* *Lines 13-17:* This second step in the `build` job is listing (using `ls -al`) the contents of the repo that has been checked out. You can now perform further actions on this repo.
+
+As before, commit and push your updated `config.yml` file.
+
+You should now see some additional steps on the CircleCI dashboard:
+
+* *Checkout code* has cloned the code from your git repository.
+* *The Second Step* has listed the files found in your git repository.
+
+image::config-second-step.png[Checking out your repo]
+
+== Part 3: Using different environments and creating workflows
+
+In Parts 1 and 2, you ran your job in basic Linux-based Docker containers.
+
+With CircleCI, you can run different jobs in different execution environments, such as macOS, Windows, virtual machines, or Docker images. By changing the Docker image, you can quickly upgrade your environment version or change languages. 
```

Comment:
WIndows and macOS _are_ Virtual machines so this needs altering slightly, also we run jobs in Docker containers not images I guess. We have this issue in lots of docs though!

Reply:
Good point! Have changed it to just virtual machines and Docker containers. I haven't included VMs in the sample code, as I don't want to overwhelm them with too many possibilities in a basic tutorial.

Reply:
Yeah good call